lib/util/host-rules.spec.ts

```diff
@@ -107,6 +110,68 @@ describe(getName(), () => {
           .token
       ).toBeUndefined();
     });
+    it('matches on specific path', () => {
+      // Initialized platform holst rule
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com',
+        token: 'defaultToken',
+      });
+      // Initialized generic host rule for github platform
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com',
+        token: 'defaultToken',
+      });
+      // specific host rule for using other token in different org
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com/repos/org-b/',
+        token: 'overwrittenToken',
+      } as any);
+      expect(
+        find({
+          hostType: PLATFORM_TYPE_GITHUB,
+          url: 'https://api.github.com/repos/org-b/someRepo/tags?per_page=100',
+        }).token
+      ).toEqual('overwrittenToken');
+    });
+    it('matches for several hostTypes when no hostType rule is configured', () => {
+      add({
+        matchHost: 'https://api.github.com',
+        token: 'abc',
+      });
+      expect(
+        find({
+          hostType: PLATFORM_TYPE_GITHUB,
+          url: 'https://api.github.com/repos/org-b/someRepo/tags?per_page=100',
+        }).token
+      ).toEqual('abc');
+      expect(
+        find({
+          hostType: 'github-releases',
+          url: 'https://api.github.com/repos/org-b/someRepo/tags?per_page=100',
+        }).token
+      ).toEqual('abc');
+    });
+    it('matches if hostType is configured and host rule is filtered with datasource', () => {
+      add({
+        hostType: PLATFORM_TYPE_GITHUB,
+        matchHost: 'https://api.github.com',
+        token: 'defaultToken',
```

Comment:
Please use shorter token to not trigger codeql hardcoded credentials issue

see https://github.com/github/codeql/blob/221a25993854d41ea455e91e146fe5ea0e8bd0ff/javascript/ql/src/semmle/javascript/security/SensitiveActions.qll#L183

for sample token / passwords